workspaces/ai-integrations/plugins/catalog-backend-module-model-catalog/src/clients/ModelCatalogGenerator.ts

```diff
@@ -189,7 +189,36 @@ export function GenerateModelServerAPI(
   };
 
   if (api.tags !== undefined) {
-    modelServerAPIEntity.metadata.tags = api.tags;
+    modelServerAPIEntity.metadata.tags = sanitizeTags(api.tags);
   }
   return modelServerAPIEntity;
 }
+
+function sanitizeMetadataName(modelName: string): string {
+  return modelName.replaceAll(/\s/g, '-');
+}
+
+function sanitizeTags(tags: string[]): string[] {
+  const sanitizedTags: string[] = [];
+  tags.forEach(tag => {
+    let sanitizedTag: string = tag;
+    // Replace whitespace with dashes
+    sanitizedTag = sanitizedTag.replace(/\s/g, '-').toLowerCase();
+
+    // Remove any invalid special characters
+    sanitizedTag = sanitizedTag.replace(/[^a-z0-9.-]/g, '-');
+
+    // Remove any successive dashes
+    sanitizedTag = sanitizedTag.replace(/-+/g, '-');
```

Comment:
yeah I'm waffling here with whether we should just skip successive dashes and log an error

Reply:
Updated